File: pkg/refined-bot/nlp_test.go
package refinedbot

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestNLPNew(t *testing.T) {
	n := NewNLP()

	intent, err := n.Parse(`lets poker XYZ-1234?`)

	assert.NoError(t, err)
	assert.NotNil(t, intent)
	assert.Equal(t, StartPokerAction, intent.Action)
}

File: refined-bot/bot_nlp/nlp_test.py
import pytest

from .nlp import NLP


def test_parse():
    issueId = 'XYZ-1234'
    text = 'lets poker %s?' % issueId

    n = NLP()
    intent = n.parse(text)

    assert intent.action == 'poker'
    assert intent.subject == issueId
    assert intent.text == text
